sshd-core/src/test/java/org/apache/sshd/common/forward/PortForwardingLoadTest.java

```diff
@@ -224,17 +224,12 @@ public void run() {
                                     }
 
                                     assertEquals("Mismatched received data at iteration #" + i, payload, baos.toString());
-
-                                    try (InputStream inputCopy = new ByteArrayInputStream(baos.toByteArray());
-                                         OutputStream sockOut = s.getOutputStream()) {
-
-                                        while (true) {
-                                            int l = sockIn.read(buf);
-                                            if (l < 0) {
-                                                break;
-                                            }
-                                            sockOut.write(buf, 0, l);
-                                        }
+                                    
+                                    //Then write it back to client
+                                    try (OutputStream sockOut = s.getOutputStream()) {
+                                        sockOut.write(baos.toByteArray(), 0, payload.length());
+                                        sockOut.flush();
+                                        sockOut.close();
```

Comment:
I probably should not have mixed this in, it is unrelated to SSHD-85, sorry. This test did not pass on the unmodified code base , it seems logically wrong, so I fixed it.